Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/LeaderBoard.hs

```diff
@@ -52,10 +63,11 @@ data LeaderBoardRes = LeaderBoardRes
 
 getDailyDriverLeaderBoard ::
   (Esq.EsqDBFlow m r, Esq.EsqDBReplicaFlow m r, EncFlow m r, Redis.HedisFlow m r, CacheFlow m r) =>
+  LConfig.DriverLeaderBoardType ->
```

Comment:
Do you really need this? We know which function is for referral and which is for ride.

Reply:
There are only two functions - getDailyDriverLeaderboard and getWeeklyDriverLeaderBoard.
getDailyDriverLeaderboard - used for two types of ranking - ride and referral
getWeeklyDriverLeaderboard - used for two types of ranking - ride and referral

if we don't pass the DriverLeaderBoardType in the function itself then we need to create 4 functions instead of two. A lot of redundant code was there so I tried to generalize the code.

Reply:
I don't remember why I added this comment earlier, it was looking redundant back then, but yeah, it looks good now